namedstruct/bits.h

```diff
@@ -139,29 +146,49 @@ namespace namedstruct {
     };
 
     /* Implementations *****************************************************************/
+
+    template <typename Int>
+    static inline typename std::make_unsigned<Int>::type zigZagEncode(Int num) {
+        static_assert(std::numeric_limits<Int>::is_integer && std::numeric_limits<Int>::is_signed);
+        using UInt = typename std::make_unsigned<Int>::type;
+        return (static_cast<UInt>(num) << 1) ^ -(static_cast<UInt>(num) >> (sizeof(Int) * 8 - 1));
+    }
+
     static inline uint32_t getLSB(uint32_t word, int numBits) {
         // TODO: use BEXTR on Intel and UBFX on ARM
         return numBits >= 32 ? word : word & ((static_cast<uint32_t>(1) << numBits) - 1);
     }
 
+    static inline const void *advance(const void *pData, int num32BitWords) {
```

Comment:
```suggestion
    static inline const void* advance(const void *pData, int num32BitWords) {
```
At least to me the old spacing was odd